samples/algorithms/sudoku-grover/Program.cs

```diff
@@ -0,0 +1,474 @@
+                Pause();
+            }
+            if (puzzleToRun == "9x9-64" || puzzleToRun == "all") 
+            {
+                // Test hard 9x9 puzzle with classical and quantum 
+                int[,] puzzle9 = {
+                    { 0,0,0, 0,0,0, 0,1,2 },
+                    { 0,0,0, 0,3,5, 0,0,0 },
+                    { 0,0,0, 6,0,0, 0,7,0 },
+                    { 7,0,0, 0,0,0, 3,0,0 },
+                    { 0,0,0, 4,0,0, 8,0,0 },
+                    { 1,0,0, 0,0,0, 0,0,0 },
+                    { 0,0,0, 1,2,0, 0,0,0 },
+                    { 0,8,0, 0,0,0, 0,4,0 },
+                    { 0,5,0, 0,0,0, 6,0,0 } };
+                int[,] puzzle9_copy = CopyIntArray(puzzle9, 9);
+                Console.WriteLine("Solving 9x9 with 64 missing numbers using classical computing");
+                ShowGrid(puzzle9, 9);
+                SolveSudukoClassic(puzzle9, 9, 3);
+                bool good = puzzle9.Cast<int>().SequenceEqual(answer9.Cast<int>());
+                if (!good)
+                    Console.WriteLine("classical test failed");
+                else
+                {
+                    Console.WriteLine("classical test passed");
+                    ShowGrid(puzzle9, 9);
+                }
+                Pause();
+                Console.WriteLine("Solving 9x9 with 64 missing numbers using Quantum Computing. Cntrl-C to stop");
+                QuantumSolve(puzzle9_copy, 9, 3, sim);
+                good = puzzle9_copy.Cast<int>().SequenceEqual(answer9.Cast<int>());
+                if (good)
+                    Console.WriteLine("quantum result verified correct");
+                Pause();
+            }
+            Console.WriteLine("finished");
+
+        }
+
+        // An Empty square with its row and column
+        public class EmptySquare
+        {
+            public int i, j;
+        }
+
+        /// # Summary
+        /// Copy an Int 2 dimensional array
+        ///
+        /// # Input
+        /// ## org
+        /// The array to copy
+        /// ## size
+        /// The size of the array
+        ///
+        /// # Output
+        /// A copy of the array
+        public static int[,] CopyIntArray(int[,] org, int size)
+        {
+            int[,] result = new int[size, size];
+            for (int i = 0; i < size; i++)
+                for (int j = 0; j < size; j++)
+                    result[i, j] = org[i, j];
+            return result;
+        }
+ 
+        /// # Summary
+        /// QuantumSolve will call Q# code to solve the Sudoku puzzle and display the solution
+        /// # Input
+        /// ## puzzle
+        /// The Sudoku puzzle to solve
+        /// ## size
+        /// The size of the puzzle is either 4 (for 4x4) or 9 (for 9x9)
+        /// ## subsize
+        /// subsize is 2 for 4x4 and 3 for 9x9
+        /// ## sim
+        /// The quantum simulator
+        public static void QuantumSolve(int[,] puzzle, int size, int subSize, QuantumSimulator sim)
```

Comment:
Since you have several `.Result` property access here, it may be easier to read as an async method.
```suggestion
        public async static Task QuantumSolve(int[,] puzzle, int size, int subSize, QuantumSimulator sim)
```